### ISSUE TITLE
bytesused vs. sizeimage

### ISSUE THREAD

**Issue by k-a-z-u** [xqms/xtion#11]
I'm not sure whether this is an issue or not. Just stumbled over it while coding and wondered whether other applications might be confused as well:

The driver reports the maximum image size to be 640 * 480 * 16bit = 614400 which is perfectly fine.
When reading images, however, the number of used bytes seems to be 565328 for every image.
If I understand correctly, this results from
```
buf->vb.v4l2_buf.bytesused = buf->pos;
vb2_set_plane_payload(&buf->vb, 0, buf->pos);
```
and somehow relates to the black bars surrounding the depth-image? Is this some sort of image-crop?

I thought different sizes were meant solely for compressed formats like JPEG, H264, etc. and not for raw formats where the image size is well-known. So: is it the drivers duty to report 614400 (instead of) 565328 and fill the remaining bytes e.g. with 0 or is it the duty of the application using the driver?

**Comment by xqms** [comment]
Thanks, I'll look into this on Wednesday when I have access to an Xtion device to try things out. I assume you are referring to the depth channel?

In that case, the driver is already introducing the "black bars" with unavailable data around the depth image. It is very much possible that `bytesused` is simply set incorrectly to the size received from the device instead of the size of the entire image buffer. I had a look at my userspace application https://github.com/xqms/xtion_grabber and it uses 640x480 buffers everywhere and does not care about the `bytesused` field.

So probably there is a bug, I'll check it on Wednesday. In the meantime, try ignoring the `bytesused` field ;-)

**Comment by k-a-z-u** [comment]
Yes, I was referring to the depth-channel.
I'm working on a little library for v4l2 in general and had funny random noise at the end of the image ;)
I had to use the `bytesused` field as some of my webcams send MJPEG where the compressed size varies for every image.

**Comment by xqms** [comment]
Hm, I even get yet another number in the `bytesused` field: 574288.

The numbers used for padding are exactly the same as in the OpenNI2 driver. The only explanation I have at this point is that we are loosing a data packet at the end. I will investigate further...

**Comment by xqms** [comment]
Finally fixed in dcb9216bf52d3e87b56ec87cb535def9385518da :-)
